internal-webapps/org.wso2.bfsi.consent.management.endpoint/src/main/java/org/wso2/bfsi/consent/management/endpoint/utils/ConsentUtils.java

```diff
@@ -0,0 +1,508 @@
+        return consentData;
+    }
+
+    /**
+     * Send authorize request in order to complete the authorize flow and get the redirect.
+     *
+     * @param consent     The approval/denial of the consent of the user
+     * @param cookies     The session cookies used in auth flow
+     * @param consentData Consent data object which contains consent information
+     * @return The redirect URI to end the authorize flow
+     */
+    public static URI authorizeRequest(String consent, Map<String, String> cookies, ConsentData consentData) {
+
+        String authorizeURL = IdentityUtil.getProperty("OAuth.OAuth2AuthzEPUrl");
+        try (CloseableHttpClient client = HTTPClientUtils.createClientWithCustomVerifier().build()) {
+
+            BasicCookieStore cookieStore = new BasicCookieStore();
+            String cookieDomain = new URI(authorizeURL).getHost();
+            for (Map.Entry<String, String> cookieValue : cookies.entrySet()) {
+                BasicClientCookie cookie = new BasicClientCookie(cookieValue.getKey(), cookieValue.getValue());
+                cookie.setDomain(cookieDomain);
+                cookie.setPath("/");
+                cookie.setSecure(true);
+                cookieStore.addCookie(cookie);
+            }
+            HttpPost authorizeRequest = new HttpPost(authorizeURL);
+            List<NameValuePair> params = new ArrayList<>();
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.HAS_APPROVED_ALWAYS, "false"));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.SESSION_DATA_KEY_CONSENT,
+                    consentData.getSessionDataKey()));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.CONSENT, consent));
+            params.add(new BasicNameValuePair(ConsentExtensionConstants.USER, consentData.getUserId()));
+            HttpContext localContext = new BasicHttpContext();
+            localContext.setAttribute(HttpClientContext.COOKIE_STORE, cookieStore);
+            UrlEncodedFormEntity entity = new UrlEncodedFormEntity(params);
+            authorizeRequest.setEntity(entity);
+            HttpResponse authorizeResponse = client.execute(authorizeRequest, localContext);
+
+            if (authorizeResponse.getStatusLine().getStatusCode() != HttpURLConnection.HTTP_MOVED_TEMP) {
+                throw new ConsentException(consentData.getRedirectURI(), AuthErrorCode.SERVER_ERROR,
+                        "Error while getting authorize redirect", consentData.getState());
+            } else {
+                //Extract the location header from the authorization redirect
+                return new URI(authorizeResponse.getLastHeader(ConsentExtensionConstants.LOCATION).getValue());
+            }
+        } catch (IOException e) {
+            log.error("Error while sending authorize request to complete the authorize flow", e);
+            return null;
```

Comment:
Have we handled this null somewhere? Couldn't see this method is used anywhere in this PR.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/9/commits/09e8d2373e219e84b2c5172f0d2e5eca48823c63